ldap/src/integration-test/java/org/springframework/security/ldap/server/UnboundIdContainerTests.java

```diff
@@ -77,4 +84,97 @@ public void afterPropertiesSetWhenPortIsZeroThenRandomPortIsSelected() throws Ex
+		UnboundIdContainer server = new UnboundIdContainer(validRootDn, validLdifClassPath);
+		server.setPort(0);
+		try {
+			server.afterPropertiesSet();
+			assertThat(server.getPort()).isNotEqualTo(0);
+		} catch (Exception e) {
+			assertThat(e.getCause()).isInstanceOf(NullPointerException.class);
+			assertThat(e.getMessage()).contains("Unable to load LDIF test-server-malformed.txt");
+		} finally {
+			server.destroy();
+		}
+	}
+
+	@Test
+	public void testInvalidLdapFile() {
+		UnboundIdContainer server = new UnboundIdContainer(validRootDn, "missing-file.ldif");
+		server.setApplicationContext(new GenericApplicationContext());
+		server.setPort(0);
+		try {
+			server.afterPropertiesSet();
+		} catch (Exception e) {
+			assertThat(e.getCause()).isInstanceOf(IllegalStateException.class);
+			assertThat(e.getMessage()).contains("Unable to load LDIF test-server-malformed.txt");
+		} finally {
+			server.destroy();
+		}
+	}
+
+	@Test
+	public void testInvalidNullLdapFile() {
+		UnboundIdContainer server = new UnboundIdContainer(validRootDn, null);
+		BuildInvalidServer(server);
+	}
+
+	@Test
+	public void testInvalidEmptyPathLdapFile() {
+		UnboundIdContainer server = new UnboundIdContainer(validRootDn, "");
+		BuildInvalidServer(server);
+	}
+
+	@Test
+	public void testInvalidDirectoryAsLdapFile() {
+		UnboundIdContainer server = new UnboundIdContainer(validRootDn, "classpath:/");
+		server.setApplicationContext(new GenericApplicationContext());
+		server.setPort(0);
+		try {
+			server.afterPropertiesSet();
+		} catch (Exception e) {
+			assertThat(e.getCause()).isInstanceOf(IllegalArgumentException.class);
+			assertThat(e.getMessage()).contains("Unable to load LDIF /");
+		} finally {
+			server.destroy();
+		}
+	}
+
+	@Test
+	public void testInvalidFileTextFileAsLdapFile() {
+		UnboundIdContainer server = new UnboundIdContainer(validRootDn, "classpath:test-server-malformed.txt");
+		BuildInvalidServer(server);
+	}
+
+	@Test
+	public void testInvalidRootDnLdapFile() {
+		UnboundIdContainer server = new UnboundIdContainer("dc=fake,dc=org", validLdifClassPath);
+		server.setApplicationContext(new GenericApplicationContext());
+		server.setPort(0);
+		try {
+			server.afterPropertiesSet();
+
+		} catch (Exception e) {
+			assertThat(e.getCause()).isInstanceOf(LDAPException.class);
+			assertThat(e.getMessage()).contains("Unable to add entry 'ou=groups,dc=springframework,dc=org' because it is not within any of the base DNs configured in the server.");
+		} finally {
+			server.destroy();
+		}
+	}
+
+	private void BuildInvalidServer(UnboundIdContainer server) {
+		server.setApplicationContext(new GenericApplicationContext());
+		server.setPort(0);
+		try {
+			server.afterPropertiesSet();
+
+		} catch (Exception e) {
+			assertThat(e.getCause()).isInstanceOf(IllegalArgumentException.class);
+			assertThat(e.getMessage()).contains("Unable to load LDIF test-server-malformed.txt");
+		} finally {
+			server.destroy();
+		}
+	}
+
```

Comment:
Please add a test for the original issue.
The test should create an `UnboundIdContainer` with an LDIF that does not exist, and ensure that an exception is thrown.